lsm6ds3tr/lsm6ds3tr.go

```diff
@@ -84,30 +83,32 @@ func (d *Device) doConfigure(cfg Configuration) (err error) {
 		d.gyroSampleRate = GYRO_SR_104
 	}
 
-	data := d.buf[:1]
-
 	// Configure accelerometer
-	data[0] = uint8(d.accelRange) | uint8(d.accelSampleRate)
-	err = legacy.WriteRegister(d.bus, uint8(d.Address), CTRL1_XL, data)
+	d.buf[0] = CTRL1_XL
+	d.buf[1] = uint8(d.accelRange) | uint8(d.accelSampleRate)
+	err = d.bus.Tx(d.Address, d.buf[0:2], nil)
 	if err != nil {
 		return
 	}
 
 	// Set ODR bit
-	err = legacy.ReadRegister(d.bus, uint8(d.Address), CTRL4_C, data)
+	d.buf[0] = CTRL4_C
+	err = d.bus.Tx(d.Address, d.buf[0:1], d.buf[1:2])
 	if err != nil {
 		return
 	}
-	data[0] = data[0] &^ BW_SCAL_ODR_ENABLED
-	data[0] |= BW_SCAL_ODR_ENABLED
-	err = legacy.WriteRegister(d.bus, uint8(d.Address), CTRL4_C, data)
+	d.buf[0] = CTRL4_C
+	d.buf[1] = d.buf[1] &^ BW_SCAL_ODR_ENABLED
+	d.buf[1] |= BW_SCAL_ODR_ENABLED
+	err = d.bus.Tx(d.Address, d.buf[0:2], nil)
 	if err != nil {
 		return
 	}
 
 	// Configure gyroscope
-	data[0] = uint8(d.gyroRange) | uint8(d.gyroSampleRate)
-	err = legacy.WriteRegister(d.bus, uint8(d.Address), CTRL2_G, data)
+	d.buf[0] = CTRL2_G
+	d.buf[1] = uint8(d.gyroRange) | uint8(d.gyroSampleRate)
+	err = d.bus.Tx(d.Address, d.buf[0:2], nil)
```

Comment:
Could we abstract these calls to something like `d.writeReg8/readReg8` ?

```go
func (d *Device) writeReg8(addr, value byte) error {
   d.buf[0] = addr
   d.buf[1] = value
   return err = d.bus.Tx(d.Address, d.buf[0:2], nil)
}

func (d *Device) readReg8(addr byte) (byte,error) {
   d.buf[0] = addr
   err := d.bus.Tx(d.Address, d.buf[0:1], d.buf[1:2])
   return d.buf[1], err
}
```

Reply:
This is just one of many drivers that scream for such improvement, a trial balloon if you wish.
In other words, each of these drivers will have to repeat the pattern. Any ideas how could we implement these helper functions once and re-use them? Instead of implementing in every driver anew?

Reply:
@soypat @aykevl I'm fine with adding helper functions. Shall I just add them and replicate the pattern in other drivers? Or is there a better way?

Reply:
I've created [regmap](https://github.com/tinygo-org/drivers/pull/768/files). Thanks for the suggestion